xarray/backends/netCDF4_.py

```diff
@@ -156,10 +156,12 @@ def _extract_nc4_variable_encoding(variable, raise_on_invalid=False,
     if lsd_okay:
         valid_encodings.add('least_significant_digit')
 
-    if (encoding.get('chunksizes') is not None and
-            (encoding.get('original_shape', variable.shape) !=
-                variable.shape) and not raise_on_invalid):
-        del encoding['chunksizes']
+    if not raise_on_invalid and 'chunksizes' in encoding:
+        chunks_too_big = any(
+            c > d for c, d in zip(encoding['chunksizes'], variable.shape))
```

Comment:
It looks like you'll need to make sure `encoding['chunksizes']` is an iterable of length `variable.ndim`